src/daft-stats/src/table_stats.rs

```diff
@@ -104,6 +116,21 @@ impl TableStatistics {
             _ => Ok(ColumnRangeStatistics::Missing),
         }
     }
+
+    pub fn cast_to_schema(&self, schema: SchemaRef) -> crate::Result<TableStatistics> {
```

Comment:
As posted above, I dont believe that casting preserves min / max

Reply:
Changed `ColumnStats::cast` to be stricter about what types it can handle